src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtils.java

```diff
@@ -0,0 +1,176 @@
+                break;
+            case POSTPONEMENT:
+                pdfUrlBuilder.append("Postponement Request");
+                handlePostponement(sscsCaseData);
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported event type for processRequestPdfAndSetPreviewDocument: " + pdfType);
+        }
+        pdfUrlBuilder.append(".pdf");
+
+        if (isBlank(requestDetails)) {
+            response.addError(String.format("Please enter request details to generate a %s document", pdfType.toString().toLowerCase()));
+            return response;
+        }
+
+        DocumentLink previewDocument = getPreviewDocument(pdfUrlBuilder.toString(), userAuthorisation, generateFile, templateId, title);
+
+        switch (pdfType) {
+            case POST_HEARING:
+                sscsCaseData.getDocumentStaging().setPreviewDocument(previewDocument);
+                break;
+            case POSTPONEMENT:
+                sscsCaseData.getPostponementRequest().setPostponementPreviewDocument(previewDocument);
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported event type for processRequestPdfAndSetPreviewDocument: " + pdfType);
+        }
+
+        return response;
+    }
+
+    private static void handlePostponement(SscsCaseData sscsCaseData) {
+        requestDetails = sscsCaseData.getPostponementRequest().getPostponementRequestDetails();
+        String hearingVenue = sscsCaseData.getPostponementRequest().getPostponementRequestHearingVenue();
+        LocalDate hearingDate = LocalDateTime.parse(sscsCaseData.getPostponementRequest().getPostponementRequestHearingDateAndTime()).toLocalDate();
+        ADDITIONAL_REQUEST_DETAILS.append("Date of Hearing: ").append(hearingDate.format(DATE_TIME_FORMATTER)).append("\n");
+        ADDITIONAL_REQUEST_DETAILS.append("Hearing Venue: ").append(hearingVenue).append("\n");
+        ADDITIONAL_REQUEST_DETAILS.append("Reason for Postponement Request: ").append(requestDetails).append("\n");
+        title = "Postponement Request from FTA";
+    }
+
+    private static void handlePostHearing(SscsCaseData sscsCaseData) {
+        requestDetails = sscsCaseData.getDocumentGeneration().getBodyContent();
+        LocalDate issueFinalDecisionDate = sscsCaseData.getIssueFinalDecisionDate();
+        if (issueFinalDecisionDate == null) {
+            throw new IllegalArgumentException("issueFinalDecisionDate unexpectedly null for caseId: " + sscsCaseData.getCcdCaseId());
+        }
+        ADDITIONAL_REQUEST_DETAILS.append("Date of decision issued: ").append(issueFinalDecisionDate.format(DATE_TIME_FORMATTER)).append("\n");
+        String postHearingRequestType = sscsCaseData.getPostHearing().getRequestType().getDescriptionEn();
+        ADDITIONAL_REQUEST_DETAILS.append("Reason for ").append(postHearingRequestType).append(" request: ").append(requestDetails).append("\n");
```

Comment:
append when its longer drop it down like
```.append("Reason for ")
.append(postHearingRequestType)
.append(" request: ")
.append(requestDetails)
.append("\n");```

Reply:
done